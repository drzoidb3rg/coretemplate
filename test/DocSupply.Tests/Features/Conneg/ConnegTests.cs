using System;
using System.IO;
using Newtonsoft.Json;
using NUnit.Framework;
using Shouldly;
using Shouldly.Configuration;


namespace DocSupply.Tests
{


    [TestFixture]
    public class ConnegTests
    {
        public ConnegTests()
        {
           WebTest.WithDiffTool();
        }

        [TestCase]
        public void ConnegTests_GetJson()
        {
            using (var _ = Tester.AsUserOne()
                .AsJson()
                .Get("/conneg/1")
                .IsOk()
                .IsJsonContent()) { }
        }

        [TestCase]
        public void ConnegTests_GetHtml()
        {
            using (var _ = Tester.AsUserOne()
                .AsHtml()
                .Get("/conneg/1")
                .IsOk()
                .IsHtmlContent())
            {
                _.ResponseString().ShouldMatchApproved(_.WithApprovalFolder);
            }
        }

        [TestCase]
        public void ConnegTests_PutAsJson()
        {
            using (var _ = Tester.AsUserOne()
                .AsJson()
                .Put("/conneg/1", new HttpForm().Add("Title", "[new title]").AsFormContent())
                .IsOk()
                .IsJsonContent())
            {
                _.PrettyJsonResponse().ShouldMatchApproved(_.WithApprovalFolder);
            }
        }
    }
}
