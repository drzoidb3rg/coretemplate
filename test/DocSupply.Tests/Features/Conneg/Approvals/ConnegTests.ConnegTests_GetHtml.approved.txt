
<p><h2>some title</h2></p>

<p><a href="/conneg/1">Link to some title</a></p>



    <form action="/conneg/1" method="POST">
        Title:<br>
        <input type="text" name="Title"><br>

        <input type="hidden" name="_method" value="put" />
        <input type="submit" value="Submit">
    </form>
